invokeai/app/invocations/denoise_latents.py

```diff
@@ -498,6 +499,33 @@ def parse_controlnet_field(
                 )
             )
 
+    @staticmethod
+    def parse_t2i_adapter_field(
+        exit_stack: ExitStack,
+        context: InvocationContext,
+        t2i_adapters: Optional[Union[T2IAdapterField, list[T2IAdapterField]]],
+        ext_manager: ExtensionsManager,
+    ) -> None:
+        if t2i_adapters is None:
+            return
+
+        # Handle the possibility that t2i_adapters could be a list or a single T2IAdapterField.
+        if isinstance(t2i_adapters, T2IAdapterField):
+            t2i_adapters = [t2i_adapters]
+
+        for t2i_adapter_field in t2i_adapters:
+            ext_manager.add_extension(
+                T2IAdapterExt(
+                    node_context=context,
```

Comment:
Hmm... I don't love that we have to pass the InvocationContext to the backend. It feels like we are mixing concerns, and increasing the likelihood of circular imports. But, I can't immediately think of a better way. Let me know if you had any other ideas, otherwise we'll leave it as-is.

Reply:
I don't know other ways to allow extensions load-unload models.
It will be used same way in ip-adapter PR.